### ISSUE TITLE
Intermittent issue - build stage in OSIO and Jenkins are not in sync

### ISSUE THREAD

**Issue by ljelinkova** [openshiftio/openshift.io#4172]
##### Issue Overview
I created simple application from booster and build it successfully for the first time. During second build I encountered strange state when OSIO showed stage 'Build release' as failed but the Jenkins log showed that the application successfully deployed to stage and prompted user to promote to run. 

![screenshot from 2018-08-15 11-13-28](https://user-images.githubusercontent.com/2078072/44144292-28d04974-a087-11e8-9497-5e49f2ad8cc5.png)

![screenshot from 2018-08-15 11-13-52](https://user-images.githubusercontent.com/2078072/44144300-31d29a4a-a087-11e8-8bf9-8a4d3fa573ba.png)

After some time I could see "failed" mark at the beginning of the Jenkins job but the job was still running asking to promote application to run. However, clicking the link had no effect. 

![screenshot from 2018-08-15 11-13-42](https://user-images.githubusercontent.com/2078072/44144423-bab6912c-a087-11e8-81a6-f5db69067960.png)

And when I looked into "master" I noticed that the second job is not listed at all but that there is some 3rd job. 

![screenshot from 2018-08-15 11-15-51](https://user-images.githubusercontent.com/2078072/44144437-c73cf148-a087-11e8-859d-dbe6cddd5964.png)

However, I can see the job when I show the Jenkins dashboard. 

![screenshot from 2018-08-15 11-15-41](https://user-images.githubusercontent.com/2078072/44144896-a6b7d6d4-a089-11e8-9edd-a932f97e3950.png)

This is the Jenkins log
[log.txt](https://github.com/openshiftio/openshift.io/files/2289968/log.txt)

##### Additional Information
I've seen similar issue in E2E tests - although previously the job did not fail - OSIO showed that Build/Release stage is running and Jenkins log was at the Promt for rolout to stage. It stayed that way until E2E tests timed out after 30 minutes.

**Comment by ppitonak** [comment]
I found two jobs on prod-preview in last 24h with this bug:

* http://artifacts.ci.centos.org/devtools/e2e/devtools-test-e2e-prod-preview.openshift.io-smoketest-pr-us-east-2a-beta/1538
* http://artifacts.ci.centos.org/devtools/e2e/devtools-test-e2e-prod-preview.openshift.io-smoketest-pr-us-east-2a-released/1557/

**Comment by ljelinkova** [comment]
This issue started to occur on pretty frequent basis approximately in the last 7 days. At first, the build starts in OSIO as number 2 but once it is finished the OSIO UI changes to build number 1. 

Are there any additional information, logs etc. that we could provide?
These are screenshots from e2e tests:

1. Build number 2 is started
![05-01-pipeline-info](https://user-images.githubusercontent.com/2078072/46080090-3e0d0d80-c199-11e8-8229-8fc4cbd23153.png)

2. Build number 2 is finished

![05-05-pipeline-finished](https://user-images.githubusercontent.com/2078072/46080111-4e24ed00-c199-11e8-9777-d4407f64108b.png)

3. Jenkins log
http://artifacts.ci.centos.org/devtools/e2e/devtools-test-e2e-openshift.io-smoketest-us-east-2-released/753/05-06-jenkins-log.html
![05-06-jenkins-log](https://user-images.githubusercontent.com/2078072/46080159-701e6f80-c199-11e8-9164-a35cc2b052da.png)

4. OS console
![05-07-os-pipeline](https://user-images.githubusercontent.com/2078072/46080173-7dd3f500-c199-11e8-97aa-21964ab3d4e7.png)

5. OSIO Pipelines screen
![05-08-aftereach](https://user-images.githubusercontent.com/2078072/46080189-86c4c680-c199-11e8-9a2b-9e779045c328.png)


These are some of oc logs
http://artifacts.ci.centos.org/devtools/e2e/devtools-test-e2e-openshift.io-smoketest-us-east-2-released/753/oc-logs-output.txt

**Comment by ppitonak** [comment]
@joshuawilson can somebody from your team have a look?

**Comment by joshuawilson** [comment]
Yes, we can look at it.

**Comment by piyush-garg** [comment]
This issue seems like the side effect of #3742 to me. Because in that case the second job of Jenkins get sync to first build of pipeline and the first job of Jenkins gets sync to the second build of the pipeline. One job succeeds and one job fails and it appears opposite to us in OSIO pipeline. Just a thought as I have observed this behavior.

**Comment by joshuawilson** [comment]
I've added an story to track this at https://openshift.io/openshiftio/Openshift_io/plan/detail/802

**Comment by alexeykazakov** [comment]
I've removed "platform" label since it looks like a build issue.

**Comment by rupalibehera** [comment]
I tried to reproduce this 7-8 times, it did not reproduce at all, we suspect that this issue might be related to the launcher fix for this https://github.com/openshiftio/openshift.io/issues/3742

**Comment by rupalibehera** [comment]
@ppitonak , I think we can reduce the priority of this issue as it is so intermittent.

**Comment by ppitonak** [comment]
@rupalibehera what about decreasing the priority **after** you know the root cause? Are you 100% sure that it's a duplicate of #3742? We did experience #3742 recently.

**Comment by rupalibehera** [comment]
> @rupalibehera what about decreasing the priority **after** you know the root cause? Are you 100% sure that it's a duplicate of #3742? We did experience #3742 recently.

@ppitonak , I was not able to reproduce it, finding the root cause becomes difficult them. 
I am not 100% sure if that is a duplicate of https://github.com/openshiftio/openshift.io/issues/3742, it can be a side effect though so lets wait for the fix from launcher team for https://github.com/openshiftio/openshift.io/issues/3742.

**Comment by piyush-garg** [comment]
@rupalibehera @ppitonak @sthaha 

I was able to reproduce this issue as I was testing some other fix.

I observed the same behavior as mentioned earlier. Two builds get triggered as the initial setup that is #3742

After that second build on osio get sync to the first job in Jenkins and first build of osio get sync to the second build of Jenkins. Sometimes both pass or fail. Appears opposite to us. Clicking on `View Log` also redirects to opposite Jenkins build job like mentioned earlier.

Attaching screenshot

![osio-two](https://user-images.githubusercontent.com/19270240/47747184-c194af80-dcad-11e8-9ccb-b2fae605c288.png)

![jenkins-2](https://user-images.githubusercontent.com/19270240/47747193-c9545400-dcad-11e8-8fb4-123897a371e4.png)

cc @tinakurian @bartoszmajsak

**Comment by ldimaggi** [comment]
How were you able to reproduce the error? Manually - or with an automated test?

**Comment by ppitonak** [comment]
https://ci.centos.org/job/devtools-test-e2e-openshift.io-smoketest-us-east-2-released/964/  (Oct 31, 2018 10:05:00 AM)

**Comment by ppitonak** [comment]
https://ci.centos.org/job/devtools-test-e2e-prod-preview.openshift.io-smoketest-pr-us-east-2a-released/3553/console

**Comment by ppitonak** [comment]
@rupalibehera can you post update? The issue has been open for almost 3 months.

**Comment by piyush-garg** [comment]
@ppitonak I already commented my findings there https://github.com/openshiftio/openshift.io/issues/4172#issuecomment-434451200 and its the side effect of #3742 

@rupalibehera mentioned earlier also https://github.com/openshiftio/openshift.io/issues/4172#issuecomment-431794196

Thanks

**Comment by ppitonak** [comment]
OK, I asked @tinakurian for update on #3742

**Comment by rupalibehera** [comment]
@ppitonak do you still this issue ?
